### DISCUSSION TITLE
no recording metrics from ruler

### DISCUSSION THREAD

**Discussion by arnitolog** [grafana/mimir#6989] [Help and support] [answered]
Hello,

I have Mimir Ruler up and running with Azure blob as storage. in the storage, some folders were created, which makes me believe that Ruler is able to calculate recording rules successfully. 
However, I'm not able to query these metrics in Grafana.
So, it is not clear which component should read metrics from the ruler's blob container.
Should it be a dedicated store gateway or a ruler itself?

**Comment by dimitarvdimitrov** [comment, marked as answer]
The ruler writes the metrics from rules back to the ingesters and they end up in the blocks storage bucket. The ruler only uses a bucket to store the definitions of the rules. This means that you can query the metrics from rules just like any other metric. 

One simple reason for why you might not be seeing the results is that the rules are running under a different tenant than what the datasource in Grafana is set up with. 

Another step is to check whether running the rules' queries yourself yields any results or do you  get "No data" when running them in Grafana?

**Comment by arnitolog** [comment]
yeah, it looks like the tenant's name is the root cause of my issue. Thanks a lot, @dimitarvdimitrov
